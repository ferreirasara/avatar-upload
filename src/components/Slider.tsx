import '../style/Slider.scss';
import { SliderProps } from "../@types/Slider";

export const Slider = (props: SliderProps) => {
  const { onChange, value } = props;
  return <input
    type="range"
    className='slider'
    value={value}
    max={100}
    min={0}
    data-testid={'slider'}
    style={{ backgroundSize: value * 100 / 100 + '% 100%' }}
    onChange={(e) => onChange(parseInt(e?.target?.value))}
  />
}