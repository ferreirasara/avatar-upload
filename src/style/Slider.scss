.slider {
  -webkit-appearance: none;
  width: 100%;
  height: 2px;
  outline: none;
  -webkit-transition: .2s;
  transition: opacity .2s;
  background: #B9D1FF;
  border-radius: 1px;
  background-image: linear-gradient(#3F80FF, #3F80FF);
  background-size: 50% 100%;
  background-repeat: no-repeat;
}

.slider::-webkit-slider-thumb {
  -webkit-appearance: none;
  appearance: none;
  width: 12px;
  height: 12px;
  border-radius: 50%;
  background: #3F80FF;
  cursor: pointer;
}

.slider::-moz-range-thumb {
  width: 12px;
  height: 12px;
  border-radius: 50%;
  background: #3F80FF;
  cursor: pointer;
}