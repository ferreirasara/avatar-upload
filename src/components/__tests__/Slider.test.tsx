import renderer from 'react-test-renderer';
import { Slider } from '../Slider';
import { render, fireEvent } from '@testing-library/react';

const onChange = jest.fn();
describe('Slider', () => {
  it('match snapshot', () => {
    const slider = renderer.create(<Slider value={50} onChange={onChange} />).toJSON();
    expect(slider).toMatchSnapshot();
  });

  it('call onChange function', () => {
    const { getByTestId } = render(
      <Slider value={50} onChange={onChange} />
    );
    fireEvent.change(getByTestId('slider'), { target: { value: 25 } });
    expect(onChange).toHaveBeenCalledWith(25);
  });
})