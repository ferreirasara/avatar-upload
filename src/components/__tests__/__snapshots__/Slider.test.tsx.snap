// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider match snapshot 1`] = `
<input
  className="slider"
  data-testid="slider"
  max={100}
  min={0}
  onChange={[Function]}
  style={
    Object {
      "backgroundSize": "50% 100%",
    }
  }
  type="range"
  value={50}
/>
`;
